### ISSUE TITLE
trap_method (and trap_class_method) not sending blocks

### ISSUE THREAD

**Issue by markiz** [sporkrb/spork#30]
Imagine, I've got to trap Spec::Runner's .configure method. It takes a block whose execution I need to delay. But as of now, spork (I've got 0.8.1) doesn't resend blocks.

As far as I can see, it's not that hard to implement block resending, I could even do it myself if needed, although I'm not that familiar with spork internals to say it'll go without consequences.

**Comment by timcharper** [comment]
Markiz: is it due to the prevent_double_run functionality?  I wrestled with whether or not that would be a good default, and after the fight concluded that it would be better than the alternative (there are a lot of problems that are introduced by blocks being run multiple times when it shouldn't).

Are you using Spork.each_run were you using Spork.trap_method?

If using the prior, try passing (false) to Spork.each_run

**Comment by markiz** [comment]
Sorry, I've long forgotten about this thread.
The general idea was that
    class SomeClass
      def self.foo(bar)
        yield(bar)
      end
    end

```
Spork.trap_class_method(SomeClass, :foo)
SomeClass.foo("Ho-ho-ho!") do |bar|
  puts bar
end
```

The above code imho should work as intended but because of the implementation:
        def #{method_name}(_args)
          Spork.each_run(false) do
            #{method_name_without_spork}(_args)
          end
        end
        # should accept and send &block, obviously

it doesn't

**Comment by timcharper** [comment]
I pushed a fix. It will be released in spork 0.8.5.
